task3/checkpoint_fsync.c

```diff
@@ -94,11 +94,20 @@ init_matrix (double** matrix)
 	}
 }
 
-inline long nano_diff(long l, long r) {
-	if (l > r) {
-		return l - r;
+inline unsigned long long nano_diff(struct timespec l, struct timespec r) {
+	unsigned long long foo;
+	if (r.tv_sec > l.tv_sec) {
+		foo = ((unsigned long long)(r.tv_sec - l.tv_sec)) * SEC_TO_NANO;
+	} else {
+		foo = ((unsigned long long)(l.tv_sec - r.tv_sec)) * SEC_TO_NANO;
 	}
-	return r - l;
+	unsigned long long rest;
+	if (l.tv_nsec > r.tv_nsec) {
+		rest = l.tv_nsec - r.tv_nsec;
+	} else {
+		rest = r.tv_nsec - l.tv_nsec;
+	}
+	return foo + rest;
```

Comment:
But that is not how time works. Lets make an example, for simplicity assume `SEC_TO_NANO = 1000` : `l = {tv_sec = 0, tv_nsec = 200}` and `r = {tv_sec = 1, tv_nsec = 100}`. We should see that the diff are 900 nanos. 
This method currently makes `foo=1000` and then computes `rest=100`. So the overall time is 1100, which is just wrong.
It should always be `end timestamp - start timestamp`. The rest can and should be signed.

Reply:
True, yeah sorry was in a hurry there and totally did something else, you are absolutely right. Gonna correct this right away.